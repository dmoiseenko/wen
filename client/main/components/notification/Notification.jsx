import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';


function Notification({
  description,
  remove,
  id,
  isError,
  isEvent
}) {
  function handleClick() {
    remove(id);
  }

  const notificationClassName = classNames(
    'notification',
    { 'is-warning': isError },
    { 'is-primary': isEvent }
  );

  return (
    <div
      className={notificationClassName}
      onClick={handleClick}
    >
      {description}
    </div>
  );
}

Notification.propTypes = {
  description: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  remove: PropTypes.func.isRequired,
  isError: PropTypes.bool.isRequired,
  isEvent: PropTypes.bool.isRequired
};

export default Notification;
