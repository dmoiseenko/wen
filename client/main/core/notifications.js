import uuid from 'uuid';

export {
  NOTIFICATION_ERROR,
  NOTIFICATION_EVENT,
} from './notification.constants';


export const notificationTimeout = 5000;

export function compose(description, type) {
  return {
    id: uuid.v4(),
    description,
    type,
  };
}

export function getNotificationIndex(notifications, id) {
  return notifications.findIndex(notification => notification.id === id);
}
