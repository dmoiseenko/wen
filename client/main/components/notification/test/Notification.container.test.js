import * as container from '../Notification.container';
import * as module from '../../../redux/modules/notifications.module';


it('should map state to props correctly', () => {
  const ownProps = {
    notification: {
      description: 'notification description',
      id: 1,
      type: 'event'
    }
  };

  const actual = container.mapStateToProps({}, ownProps);

  expect(actual).toMatchSnapshot();
});

it('should map dispatch to props correctly', () => {
  const actual = container.mapDispatchToProps;

  expect(actual).toEqual({
    remove: module.removeNotification
  });
});
