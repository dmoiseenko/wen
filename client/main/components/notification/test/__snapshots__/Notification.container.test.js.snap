// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should map state to props correctly 1`] = `
Object {
  "description": "notification description",
  "id": 1,
  "isError": false,
  "isEvent": true,
  "type": "event",
}
`;
